source/agent/addons/quic/QuicTransportStream.cc

```diff
@@ -200,14 +200,16 @@ void QuicTransportStream::SignalOnData()
         return;
     }
 
-    auto readableBytes = m_stream->ReadableBytes();
-    uint8_t* buffer = new uint8_t[readableBytes];
-    owt_base::Frame frame;
-    frame.format = owt_base::FRAME_FORMAT_DATA;
-    frame.length = readableBytes;
-    frame.payload = buffer;
-    m_stream->Read(frame.payload, readableBytes);
-    deliverFrame(frame);
+    while (m_stream->ReadableBytes() > 0) {
+        auto readableBytes = m_stream->ReadableBytes();
+        uint8_t* buffer = new uint8_t[readableBytes];
+        owt_base::Frame frame;
```

Comment:
Quic might require allocating a quite large buffer. Do allocation check here.

Reply:
Are exceptions enabled in this repo?

Reply:
No. I mean you need to add some check here to make static code scan tool happy.